contents/docs/plugins/build.md

```diff
@@ -78,9 +79,24 @@ Most options in this file are self-explanatory, but there are a few worth explor
 
 `main` determines the entry point for your plugin, where your `setupPlugin` and `processEvent` functions are. More on these later.
 
-#### type (config -> param -> type)
+#### config
 
-The type of a parameter in the config can be either `string` or `attachment`. If the type is set to `attachment`, PostHog will prompt the user for a file upload during the configuration step.
+`config` consists of an array of objects that each pertain to a specific configuration field or markdown explanation for your plugin.
+
+Each object in a config can have the following properties:
+
+|   Key    |                    Type                    |                                                                           Description                                                                           |
+| :------: | :----------------------------------------: | :-------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   type   | `"string"` or `"attachment"` or `"choice"` | Determines the type of the field - "attachment" asks the user for an upload, and "choice" requires the config object to have a `choices` array, explained below |
+|   key    |                  `string`                  |                                     The key of the plugin config field, used to reference the value from inside the plugin                                      |
+|   name   |                  `string`                  |                                          Displayable name of the field - appears on the plugin setup in the PostHog UI                                          |
+| default  |                  `string`                  |                                                                   Default value of the field                                                                    |
+|   hint   |                  `string`                  |                                             More information about the field, displayed under the in the PostHog UI                                             |
+| markdown |                  `string`                  |                                                             Markdown to be displayed with the field                                                             |
```

Comment:
a valid object in the config array is also { "markdown": "random markdown" }. In this case no type or key or anything else is needed. This can be placed between other fields. Might be worth mentioning that somehow